src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,65 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int OPERATION = 0;
+    private static final int AMOUNT = 1;
+    private static final String SEPARATOR = ",";
+    private static final String LINE_SEPARATOR = "@";
+    private static final String KEYWORD_ONE = "buy";
+    private static final String KEYWORD_TWO = "supply";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String data = readFile(fromFileName);
+        String report = createReport(data);
+        writeToFile(report, toFileName);
+    }
+
+    public String readFile(String fromFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String readedLine = bufferedReader.readLine();
+            while (readedLine != null) {
+                stringBuilder.append(readedLine).append(LINE_SEPARATOR);
+                readedLine = bufferedReader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from the file " + fromFileName, e);
+        }
+        return stringBuilder.toString();
+    }
+
+    public String createReport(String data) {
+        int buy = 0;
+        int supply = 0;
+        String []line = data.split(LINE_SEPARATOR);
```

Comment:
```suggestion
        String[] pieces = data.split(LINE_SEPARATOR);
```
auto format your file before committing - use Ctrl Alt L for Windows